packages/react-app/src/lib/gasPrice.js

```diff
@@ -94,20 +97,39 @@ class GasPriceStore {
     }
     return utils.parseUnits('50', 'gwei').toHexString();
   }
+
+  async getMedianHistoricalPrice() {
+    const response = await axios.get(
+      `https://ethgas.watch/api/gas/trend?hours=168`,
```

Comment:
Today, the ethgas.watch maintainer changed their allowed-origins to be more permissive so I used their url instead of https://mainnet.blockscout.com/ethgas.